boot.c

```diff
@@ -44,7 +44,12 @@ remap_kernel_space(uintptr_t runtime_base,
                    uintptr_t runtime_size)
 {
   /* eyrie runtime is supposed to be smaller than a megapage */
+
+  #if __riscv_xlen == 64
   assert(runtime_size <= RISCV_GET_LVL_PGSIZE(2));
+  #elif __riscv_xlen == 32
+  assert(runtime_size <= RISCV_GET_LVL_PGSIZE(1));
```

Comment:
Why this was needed? Were there any issues with megapages in RV32?

Reply:
Megapages are still the same size, I needed this because of the way megapage sizes are calculated using RISCV_GET_LVL_PGSIZE and its dependence on the RISCV_PT_LEVELS macro. Alternatively, we can just define a macro that is equal to 4 MiB and use it for both versions.